### PR TITLE
Examples: Add Curved Reprojection Demo

### PR DESCRIPTION
**Pull request by zalo**
This PR adds a new Example scene demonstrating how to perform CAVE-like reprojection onto curved display surfaces (ie curved monitors).

Someone asked me recently how CAVE displays could work on curved monitors, and I realized this was the perfect opportunity to add a new bit of unintuitive projection/distortion math to the three.js example compendium.

This math generalizes to any display surface one can make a mesh for; even [Paul Bourke's Infamous Mirror Domes](http://paulbourke.net/dome/mirrordome/) and [Mirror Boxes](http://paulbourke.net/dome/mirrorbox/).   
(I can make a separate "advanced" example for this if it's desired.)

A live demo is available here:
https://raw.githack.com/zalo/three.js/feat-curved-reprojection/examples/webgl_materials_reprojection.html

The key to understanding this one is disabling the "Set FoV to Corners" checkbox.

### CODE REVIEW

File: examples/webgl_materials_reprojection.html [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+					eyeProjectionMatrix : { value: eyeCamera.projectionMatrix },
+					eyeViewMatrix       : { value: eyeCamera.matrixWorldInverse },
+					previewOnScreen        : { value: false }
+				};
+
+				material.onBeforeCompile = (shader) => {
+
+					// Vertex Shader: Set UVs to be eye-camera screen-space; Map positions to be main screen-space.
+					shader.vertexShader =
+						shader.vertexShader.slice( 0, 18 ) +
```
Comment (LeviPesin) [inline examples/webgl_materials_reprojection.html:109]: I'm afraid such constructions can break easily.

Comment (Mugen87) [reply examples/webgl_materials_reprojection.html:109]: Could you also implement the shader with a custom `ShaderMaterial`? The material itself does not have to support any features other then the reprojection and diffuse map logic.

Comment (zalo) [reply examples/webgl_materials_reprojection.html]: Ah; that's a fair point; I was a little worried about the tonemapping and encoding steps, but I found the right macros for them.

**Comment by zalo** [comment]
Actually, I might hold this one off until I get the more visually impressive mirror-dome/mirror-box demo working nicely.

**Pull request closed without merging**